extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -186,47 +204,70 @@ private void extractSwaggerUi(ResolvedArtifact artifact, Path resourceDir) throw
         }
     }
 
-    private void updateApiUrl(Path indexHtml, String openApiPath) throws IOException {
+    private void updateSwaggerUiConfig(Path indexHtml, String openApiPath) throws IOException {
         String content = new String(Files.readAllBytes(indexHtml), StandardCharsets.UTF_8);
-        String result = updateApiUrl(content, openApiPath);
+        String result = updateConfig(content, openApiPath);
         if (result != null) {
             Files.write(indexHtml, result.getBytes(StandardCharsets.UTF_8));
         }
     }
 
-    public String updateApiUrl(String original, String openApiPath) {
-
-        Matcher uriMatcher = SWAGGER_UI_DEFAULT_API_URL_PATTERN.matcher(original);
-        if (uriMatcher.matches()) {
-            return uriMatcher.replaceFirst("$1" + openApiPath + "$3");
+    private String updateConfig(String original, String openApiPath) throws JsonProcessingException {
+        Matcher configMatcher = SWAGGER_UI_CONFIG_PATTERN.matcher(original);
+        if (configMatcher.matches()) {
+            updateApiUrl(openApiPath);
+            initObjectMapper();
+            String defaultConfig = configMatcher.group(3).trim();
+            String config;
+            String html;
+            config = objectMapper.writeValueAsString(swaggerUiConfig.getSwaggerUiProps());
+            html = configMatcher.replaceFirst("$1" + buildStringConfig(defaultConfig, config) + "$5");
+            if (swaggerUiConfig.oauth.enable) {
+                html = addOauthConfig(html, swaggerUiConfig.oauth.getConfigParameters());
+            }
+            return html;
         } else {
-            log.warn("Unable to replace the default URL of Swagger UI");
+            log.warn("Unable to replace the default configuration of Swagger UI");
             return null;
         }
     }
 
-    @ConfigRoot
-    static final class SwaggerUiConfig {
-        /**
-         * The path where Swagger UI is available.
-         * <p>
-         * The value `/` is not allowed as it blocks the application from serving anything else.
-         */
-        @ConfigItem(defaultValue = "/swagger-ui")
-        String path;
-
-        /**
-         * If this should be included every time. By default this is only included when the application is running
-         * in dev mode.
-         */
-        @ConfigItem
-        boolean alwaysInclude;
-
-        /**
-         * If Swagger UI should be enabled. By default, Swagger UI is enabled.
-         */
-        @ConfigItem(defaultValue = "true")
-        boolean enable;
+    private void updateApiUrl(String openApiPath) {
+        if (!StringUtils.isEmpty(openApiPath)) {
+            final Optional<URI> openApiUri = Optional.of(URI.create(openApiPath));
+            if (!swaggerUiConfig.url.isPresent()) {
+                swaggerUiConfig.url = openApiUri;
+            }
```

Comment:
This is not a good pattern. You shouldn't update the config. Better get the method to return you whatever you want and then use it.